dfsclient/src/main/java/org/hpccsystems/dfs/client/StreamOperationMessages.java

```diff
@@ -0,0 +1,104 @@
+/*******************************************************************************
+ *     HPCC SYSTEMS software Copyright (C) 2019 HPCC Systems®.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *******************************************************************************/
+
+package org.hpccsystems.dfs.client;
+
+import java.util.HashMap;
+import java.util.Set;
+
+public class StreamOperationMessages
+{
+    public static int CODE_UNKNOWN = -1;
+    public enum MessageType
+    {
+        Error,
+        Warning,
+        Info
+    };
+
+    private HashMap<String, Integer> map = new HashMap<>();
+    private int totalMessageCount = 0;
+
+    private static String createKey(String message, MessageType type, int code, String column, String value)
+    {
+        String hash = "";
+        switch (type)
+        {
+            case Error:
+               hash = "Error: ";
+               break;
+            case Warning:
+                hash = "Warning: ";
+                break;
+            default:
+                hash = "Info: ";
+                break;
+        }
+
+        hash += " '" + message + "'";
+
+        if (column != null && !column.isEmpty())
+            hash += " ColumnName: '" + column + "'";
+
+        if (value != null && !value.isEmpty())
+            hash += " Value: '" + value + "'";
+
+        return hash;
+    }
+
+    public int getTotalMessageCount()
+    {
+        return totalMessageCount;
+    }
+
+    public int getEffectiveMessageCount()
+    {
+        return map.size();
+    }
+
+    public void addMessage(String message, MessageType type, String column)
+    {
+        addMessage(message, type, CODE_UNKNOWN, null);
+    }
+
+    public void addMessage(String message, MessageType type, int code)
+    {
+        addMessage(message, type, code, null);
+    }
+
+    public void addMessage(String message, MessageType type, int code, String column)
+    {
+        addMessage(message, type, code, column, null);
+    }
+
+    public void addMessage(String message, MessageType type, int code, String column, String value)

```

Comment:
I am slightly concerned about the performance implications of creating a hashed key for each field on something like the uint8 overflow warning; we would probably be calling addMessage() every other field. 

Maybe we can allow users to pre-calculate keys, and create an addMessage() that takes a key?

Reply:
I agree. I'm not sure if it's the same as your proposal, but I did consider pre-calculated messages based on an error code. The keys would need to be unique per code +field